hathor/p2p/netfilter/matches.py

```diff
@@ -0,0 +1,136 @@
+from ipaddress import ip_network
+from typing import TYPE_CHECKING, Any
+
+from twisted.internet.defer import Deferred
+from twisted.internet.task import LoopingCall
+
+if TYPE_CHECKING:
+    from hathor.p2p.netfilter.context import NetfilterContext
+
+
+class NetfilterMatch:
+    def match(self, context: 'NetfilterContext') -> bool:
+        raise NotImplementedError
+
+
+class NetfilterMatchAll(NetfilterMatch):
+    def match(self, context: 'NetfilterContext') -> bool:
+        return True
+
+
+class NetfilterMatchAnd(NetfilterMatch):
+    def __init__(self, a, b):
+        self.a = a
+        self.b = b
+
+    def match(self, context: 'NetfilterContext') -> bool:
+        if not self.a.match(context):
+            return False
+        if not self.b.match(context):
+            return False
+        return True
+
+
+class NetfilterMatchOr(NetfilterMatch):
+    def __init__(self, a, b):
+        self.a = a
+        self.b = b
+
+    def match(self, context: 'NetfilterContext') -> bool:
+        if self.a.match(context):
+            return True
+        if self.b.match(context):
+            return True
+        return False
+
+
+class NetfilterMatchIPAddress(NetfilterMatch):
+    def __init__(self, host):
+        self.host = ip_network(host)
+
+    def match(self, context: 'NetfilterContext') -> bool:
+        if context.addr is None:
+            return False
+
+        if not hasattr(context.addr, 'host'):
+            return False
+
+        host = ip_network(context.addr.host)
+        if not self.host.supernet_of(host):
+            return False
+
+        return True
+
+
+class NetfilterMatchPeerId(NetfilterMatch):
+    def __init__(self, peer_id):
+        self.peer_id = peer_id
+
+    def match(self, context: 'NetfilterContext') -> bool:
+        if context.protocol is None:
+            return False
+
+        if context.protocol.peer is None:
+            return False
+
+        if context.protocol.peer.id != self.peer_id:
+            return False
+
+        return True
+
+
+class NetfilterMatchPeerIdRemoteURL(NetfilterMatch):
+    def __init__(self, name, manager, url, update_interval=30) -> None:
+        self.name = name
+        self.manager = manager
+        self.reactor = manager.reactor
+        self.url = url
+        self.update_interval = update_interval
+        self.lc_update = LoopingCall(self.update)
+        self.items = set()
+        self.method = 'GET'
+        self.headers = {
+            'User-Agent': ['hathor-core'],
+        }
+
+    def start(self):
```

Comment:
These should also be annotated with `-> None`.

Reply:
Done. Thanks!